Fix the log2 function to return values

jsternberg opened the pull request:
The passed in argument wasn't correct so it always returned null instead
of the appropriate value.

Also includes unit tests for all of the math functions and restricts the
`asin()` and `acos()` functions to floats only since those functions
don't give any meaningful results when using integers or unsigned.

Fixes #9850.

mark-rushakoff commented:
Is there a test that should be added or updated here?

mark-rushakoff commented:
Are you saying you will add tests to this PR or a separate PR?

If it's this PR, I'll wait to approve. If it's a different PR that will happen later, I can approve this PR if you link back with an issue tracking the missing tests.

hercules-influx commented:
During a run of [megacheck](https://github.com/dominikh/go-tools) the following issues were discovered:
```/tmp/547485954/src/github.com/influxdata/influxdb/cmd/influx-tools/importer/importer.go:73:3: this value of rpi is never used (SA4006)
```

jsternberg commented:
@mark-rushakoff I've added the unit tests. I found an issue with `asin()` and `acos()` in the current implementation so I've updated those too and updated the description.

mark-rushakoff commented on query/math.go:
```diff
@@ -23,14 +23,59 @@ func (MathTypeMapper) MapType(measurement *influxql.Measurement, field string) i
+		case influxql.Float, influxql.Integer, influxql.Unsigned, influxql.Unknown:
+			return influxql.Float, nil
+		default:
+			return influxql.Unknown, fmt.Errorf("invalid argument type for the first argument in %s(): %s", name, args[0])
```
Looks like this should be `..., arg0)`, since earlier code implies len(args) could be zero. Occurs a couple times in this change.

jsternberg replied:
Fixed all of these.

mark-rushakoff commented on query/math.go:
```diff
@@ -23,14 +23,59 @@ func (MathTypeMapper) MapType(measurement *influxql.Measurement, field string) i
+		case influxql.Float, influxql.Integer, influxql.Unsigned, influxql.Unknown:
+			return influxql.Float, nil
+		default:
+			return influxql.Unknown, fmt.Errorf("invalid argument type for the second argument in %s(): %s", name, args[0])
```
This one should be `arg1` rather than `args[0]`?

hercules-influx commented:
During a run of [megacheck](https://github.com/dominikh/go-tools) the following issues were discovered:
```/tmp/216100182/src/github.com/influxdata/influxdb/cmd/influx-tools/importer/importer.go:73:3: this value of rpi is never used (SA4006)
```
